MainDemo.Wpf/PopupBox.xaml

```diff
@@ -0,0 +1,198 @@
+﻿<UserControl x:Class="MaterialDesignDemo.PopupBox"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:local="clr-namespace:MaterialDesignDemo"
+             xmlns:materialDesign="http://materialdesigninxaml.net/winfx/xaml/themes"
+             xmlns:system="clr-namespace:System;assembly=System.Runtime"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+  <UserControl.Resources>
+    <ResourceDictionary>
+      <!-- This is needed to avoid runtime exceptions?! Seems like this might be a bug? -->
+      <ResourceDictionary.MergedDictionaries>
+        <ResourceDictionary Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.PopupBox.xaml" />
+      </ResourceDictionary.MergedDictionaries>
+
+      <DataTemplate x:Key="ContentTemplateGrid">
+        <Grid Width="200" Height="100">
+          <TextBlock Text="Popup content in grid" HorizontalAlignment="Center" VerticalAlignment="Center">
+            <!-- This explicit style should not be needed; possibly because the MaterialDesignMultiFloatingActionPopupBox style enforces special brushes? Or perhaps an issue with correctly applying theming? -->
+            <TextBlock.Style>
+              <Style TargetType="TextBlock" BasedOn="{StaticResource MaterialDesignTextBlock}">
+                <Setter Property="TextElement.Foreground" Value="{StaticResource MaterialDesignLightForeground}" />
+              </Style>
+            </TextBlock.Style>
+          </TextBlock>
+        </Grid>
+      </DataTemplate>
+
+      <DataTemplate x:Key="ContentTemplateGridWithBackground">
+        <Grid Width="200" Height="100" Background="Fuchsia">
+          <TextBlock Text="Popup content in colored grid" HorizontalAlignment="Center" VerticalAlignment="Center" />
+        </Grid>
+      </DataTemplate>
+
+      <DataTemplate x:Key="ContentTemplateButtonStack">
+        <!-- Margin of 10 here is to "make room" for the elevation drop shadows of the buttons. Must be compensated for in some (left/right) alignment scenarios using PopupBox.PopupHorizontalOffset -->
+        <StackPanel Margin="10">
+          <Button Content="1"
+                  Opacity="0.5"
+                  ToolTip="One with custom opacity" />
+          <Button Content="2" ToolTip="Two" />
+          <Button Content="3" ToolTip="Three" />
+        </StackPanel>
```

Comment:
The `Margin` applied to this `StackPanel` is there to ensure there is enough room for the drop shadows of the buttons in the `StackPanel`.

For the default style of the `PopupBox`, this is likely not an issue as the `Popup` contains a `Card` which contains the provided content. However, for the `MaterialDesignMultiFloatingActionPopupBox` derived styles, there isn't a concept of a container like the `Card` and it is simply the provided content which is shown. So in cases like these, if the user puts in stuff with drop shadows that would normally extend the bounds of the container, he/she would need to add a `Margin` to the root container (like done here) which is large enough to encompass the child elements AND their drop shadows. This will naturally affect the positioning of the `Popup`, and in those cases the user would need to compensate for that by setting "an equal and opposite" offset on the `PopupBox` using `PopupBox.PopupHorizontalOffset` and `PopupBox.PopupVerticalOffset`.

Reply:
I did consider (and try) `ClipToBounds=False`, but that does not seem to work for a `Popup` for something like this. It is like there is a hard cut-off at the edge of the window which represents the `Popup`. This is the reason I decided to go with a solution that "reserves space" for the drop shadow.